src/TestFramework/Coverage/LineRangeCalculator.php

```diff
@@ -44,15 +44,17 @@
  */
 final class LineRangeCalculator
 {
-    public function calculateRange(Node $node): NodeLineRangeData
+    public function calculateRange(Node $originalNode): NodeLineRangeData
     {
-        $node = $this->getOuterMostArrayNode($node);
+        if ($originalNode->getAttribute(ReflectionVisitor::IS_ON_FUNCTION_SIGNATURE, false) === true) {
+            $startLine = $originalNode->getStartLine();
+            // function signature node should always be 1-line range: (start, start)
+            return new NodeLineRangeData($startLine, $startLine);
+        }
 
-        $endLine = $node->getAttribute(ReflectionVisitor::IS_ON_FUNCTION_SIGNATURE, false) === true
-            ? $node->getStartLine() // function signature node should always be 1-line range: (start, start)
-            : $node->getEndLine();
+        $outerMostArrayNode = $this->getOuterMostArrayNode($originalNode);
 
-        return new NodeLineRangeData($node->getStartLine(), $endLine);
+        return new NodeLineRangeData($outerMostArrayNode->getStartLine(), $outerMostArrayNode->getEndLine());
```

Comment:
When I first started a fix a week ago, I came to a conclusion similar to this, but all other tests were failing.
Glad to see I was in the right path. Still, the `Generator` thing makes debugging to painful :weary: 

Putting `iterator_to_array` everywhere has been tedious, I'd like to have a magic php settings to turn every `Generator` into an array immediately for debug purposes.

Reply:
> Still, the Generator thing makes debugging to painful weary

Same feelings here. I had to debug them a while ago and they just blew my mind, it's very hard. Though I didn't do any investigation on how to improve this process.

Reply:
Same here...

Reply:
It isn't that generators are at the fault but the structure we have here isn't friendly to generator-style programming.